domain/src/DomainValidator.php

```diff
@@ -3,10 +3,11 @@
 namespace Drupal\domain;
 
 use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Extension\ModuleHandlerInterface;
 use Drupal\Core\StringTranslation\StringTranslationTrait;
 use Drupal\Component\Utility\Unicode;
-use GuzzleHttp\ClientInterface;
```

Comment:
My understanding is that it is generally best to call the Interface here, which guards against classes being swapped out.

If get() is not part of the Interface, then this change is correct and we have a hard dependency.

The rest of this looks great.